emma-language/src/test/scala/org/emmalanguage/compiler/lang/core/TrampolineSpec.scala

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright © 2014 TU Berlin (emma@dima.tu-berlin.de)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.emmalanguage
+package compiler.lang.core
+
+import compiler.BaseCompilerSpec
+
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+
+import scala.util.control.TailCalls._
+
+/** A spec for the `Core.trampoline` transformation. */
+@RunWith(classOf[JUnitRunner])
+class TrampolineSpec extends BaseCompilerSpec {
+
+  import compiler._
+
+  val trampolinePipeline: u.Expr[Any] => u.Tree =
+    pipeline(typeCheck = true)(
+      Core.anf,
+      Core.dscf,
+      tree => time(Core.trampoline(tree), "trampoline"),
+      Core.anf
```

Comment:
OK, I figured it out, it is needed only for the tests in order to align the result with the shape of the expected code snippets after `lnf`.